File: litemall-admin/serve-admin.js
const http = require('http');
const httpProxy = require('http-proxy');
const fs = require('fs');
const path = require('path');
const url = require('url');

const proxy = httpProxy.createProxyServer({});
const PORT = 9527;

const server = http.createServer((req, res) => {
  // Proxy API requests
  if (req.url.startsWith('/admin')) {
    proxy.web(req, res, {
      target: 'http://localhost:8081',
      changeOrigin: true
    });
    return;
  }

  // Serve static files
  let filePath = path.join(__dirname, 'dist', req.url === '/' ? 'index.html' : req.url);
  
  // Handle client-side routing
  if (!fs.existsSync(filePath)) {
    filePath = path.join(__dirname, 'dist', 'index.html');
  }

  fs.readFile(filePath, (err, content) => {
    if (err) {
      res.writeHead(404);
      res.end('File not found');
      return;
    }

    // Set content type
    const ext = path.extname(filePath);
    let contentType = 'text/html';
    if (ext === '.js') contentType = 'text/javascript';
    else if (ext === '.css') contentType = 'text/css';
    else if (ext === '.json') contentType = 'application/json';
    else if (ext === '.png') contentType = 'image/png';
    else if (ext === '.jpg') contentType = 'image/jpg';
    else if (ext === '.svg') contentType = 'image/svg+xml';

    res.writeHead(200, { 'Content-Type': contentType });
    res.end(content, 'utf-8');
  });
});

server.listen(PORT, () => {
  console.log(`Admin server running at http://localhost:${PORT}`);
});

File: litemall-admin/serve_admin.py
#!/usr/bin/env python3
import http.server
import socketserver
import urllib.request
import urllib.parse
import os

PORT = 9527
BACKEND_URL = "http://localhost:8080"

class ProxyHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory="dist", **kwargs)
    
    def do_GET(self):
        if self.path.startswith('/admin'):
            self.proxy_request()
        else:
            # Serve static files
            if self.path == '/':
                self.path = '/index.html'
            elif not os.path.exists(os.path.join("dist", self.path.lstrip('/'))):
                self.path = '/index.html'
            super().do_GET()
    
    def do_POST(self):
        if self.path.startswith('/admin'):
            self.proxy_request()
        else:
            self.send_error(404)
    
    def proxy_request(self):
        # Forward request to backend
        try:
            url = BACKEND_URL + self.path
            headers = {key: val for key, val in self.headers.items() if key != 'Host'}
            
            # Read POST data if present
            content_length = int(self.headers.get('Content-Length', 0))
            post_data = self.rfile.read(content_length) if content_length > 0 else None
            
            # Create request
            req = urllib.request.Request(url, data=post_data, headers=headers)
            req.get_method = lambda: self.command
            
            # Make request
            response = urllib.request.urlopen(req)
            
            # Send response
            self.send_response(response.getcode())
            for header, value in response.headers.items():
                if header.lower() not in ['connection', 'transfer-encoding']:
                    self.send_header(header, value)
            self.end_headers()
            
            # Copy response body
            self.wfile.write(response.read())
            
        except Exception as e:
            print(f"Proxy error: {e}")
            self.send_error(502, f"Bad Gateway: {str(e)}")

with socketserver.TCPServer(("", PORT), ProxyHTTPRequestHandler) as httpd:
    print(f"Server running at http://localhost:{PORT}")
    httpd.serve_forever()